src/effects/backends/builtin/compressoreffect.cpp

```diff
@@ -0,0 +1,215 @@
+            "For a ratio of 4:1, one dB remains for every 4dB of input signal above the threshold. "
+            "At a ratio of 1:1 no compression is happening, as the input is exactly the output"));
+    ratio->setValueScaler(EffectManifestParameter::ValueScaler::Logarithmic);
+    ratio->setUnitsHint(EffectManifestParameter::UnitsHint::Coefficient);
+    ratio->setNeutralPointOnScale(0);
+    ratio->setRange(1.0, 4.0, 1000);
+
+    EffectManifestParameterPointer knee = pManifest->addParameter();
+    knee->setId("knee");
+    knee->setName(QObject::tr("Knee (dB)"));
+    knee->setShortName(QObject::tr("Knee"));
+    knee->setDescription(QObject::tr("The Knee knob is used to achieve a rounder compression curve"));
+    knee->setValueScaler(EffectManifestParameter::ValueScaler::Linear);
+    knee->setUnitsHint(EffectManifestParameter::UnitsHint::Coefficient);
+    knee->setNeutralPointOnScale(0);
+    knee->setRange(0.0, 4.0, 24);
+
+    EffectManifestParameterPointer attack = pManifest->addParameter();
+    attack->setId("attack");
+    attack->setName(QObject::tr("Attack (ms)"));
+    attack->setShortName(QObject::tr("Attack"));
+    attack->setDescription(QObject::tr(
+        "The Attack knob sets the time that determines how fast the compression will set in once the signal exceeds the threshold"));
+    attack->setValueScaler(EffectManifestParameter::ValueScaler::Logarithmic);
+    attack->setUnitsHint(EffectManifestParameter::UnitsHint::Millisecond);
+    attack->setRange(0, 10, 250);
+
+    EffectManifestParameterPointer release = pManifest->addParameter();
+    release->setId("release");
+    release->setName(QObject::tr("Release (ms)"));
+    release->setShortName(QObject::tr("Release"));
+    release->setDescription(QObject::tr(
+        "The Release knob sets the time that determines how fast the compressor will recover from the gain reduction once the signal falls under the threshold. "
+        "Depending on the input signal, short release times may introduce a 'pumping' effect and/or distortion"));
+    release->setValueScaler(EffectManifestParameter::ValueScaler::Integral);
+    release->setUnitsHint(EffectManifestParameter::UnitsHint::Millisecond);
+    release->setRange(0, 150, 1500);
+
+    EffectManifestParameterPointer gain = pManifest->addParameter();
+    gain->setId("gain");
+    gain->setName(QObject::tr("Output gain"));
+    gain->setShortName(QObject::tr("Gain"));
+    gain->setDescription(QObject::tr("The Output gain knob adjusts the level of the output signal after the compression was applied"));
+    gain->setValueScaler(EffectManifestParameter::ValueScaler::Linear);
+    gain->setUnitsHint(EffectManifestParameter::UnitsHint::Decibel);
+    gain->setRange(-25, 0, 25);
+
+    return pManifest;
+}
+
+CompressorGroupState::CompressorGroupState(
+        const mixxx::EngineParameters& engineParameters)
+        : EffectState(engineParameters),
+          previousMakeUpGain(0),
+          previousStateDB(0) {
```

Comment:
reorder to fix compiler error
```suggestion,
          previousStateDB(0) 
          previousMakeUpGain(0) {
```